redwood-protocol/src/commonMain/kotlin/app/cash/redwood/protocol/protocol.kt

```diff
@@ -33,15 +33,35 @@ import kotlinx.serialization.json.buildJsonArray
 import kotlinx.serialization.json.jsonArray
 import kotlinx.serialization.json.jsonPrimitive
 
+/*
+Note: The types in this file are written in a careful way such that their serializable properties
+are scalars which can be serialized directly rather than using value classes are boxed within the
+generated kotlinx.serialization code.
```

Comment:
I don't love this, but kotlinx.serialization boxes _every_ value class instance on both serialization and deserialization paths, and this avoids that without changing the public API (aside from `invoke` vs constructor).